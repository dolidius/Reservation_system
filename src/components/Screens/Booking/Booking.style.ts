import styled from "styled-components";

export const SeatsContainer = styled.div`
    width: 100%;
    height: 100%;
    padding: 2rem 5rem;
`