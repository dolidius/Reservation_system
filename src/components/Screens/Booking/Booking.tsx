import React, { useState, useEffect } from 'react';

import { connect } from 'react-redux';
import ISeatsState from '../../../interfaces/storeStates/ISeatsState';

import Seats from '../../Seats/Seats';
import Spinner from '../../Helpers/Spinner/Spinner';

import IFormValues from '../../../interfaces/IFormValues';

interface IProps {
    seatsStore: ISeatsState,
}

const Booking: React.FC<IProps> = ({ seatsStore }) => {

    const [verticalGrid, setVerticalGrid] = useState(-1);
    const [horizontalGrid, setHorizontalGrid] = useState(-1);

    useEffect(() => {
        findSeatsGrid();
    }, [seatsStore]);

    const findSeatsGrid = () => {

        let verticalGrid: number | null = null;
        let horizontalGrid: number | null = null;

        seatsStore.seats.forEach(seat => {

            const { x, y } = seat.cords;

            if (verticalGrid === null || verticalGrid < x) { 
                verticalGrid = x;
            }

            if (horizontalGrid === null || horizontalGrid < y) { 
                horizontalGrid = y;
            }

        })
 
        if (verticalGrid === null) verticalGrid = -1; 
        if (horizontalGrid === null) horizontalGrid = -1; 

        setVerticalGrid(verticalGrid);
        setHorizontalGrid(horizontalGrid);
 
    }

    if (seatsStore.loading) {
        return (
            <Spinner />
        )
    } 

    return (
        <Seats 
            gridX={verticalGrid + 1}
            gridY={horizontalGrid + 1}
            seats={seatsStore.seats}
            seatsToChoose={seatsStore.seatsToChoose}
            nextToEachOther={seatsStore.nextToEachOther}
        />
    )

}

const mapStateToProps = (state: any) => ({
    seatsStore: state.seats
})

export default connect(mapStateToProps, {})(Booking);