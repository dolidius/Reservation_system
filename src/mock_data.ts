export const seats =  [
    {
      "id": "s02",
      "cords": {
        "x": 0,
        "y": 2
      },
      "reserved": false
    },
    {
      "id": "s03",
      "cords": {
        "x": 0,
        "y": 3
      },
      "reserved": false
    },
    {
      "id": "s04",
      "cords": {
        "x": 0,
        "y": 4
      },
      "reserved": false
    },
    {
      "id": "s06",
      "cords": {
        "x": 0,
        "y": 6
      },
      "reserved": false
    },
    {
      "id": "s07",
      "cords": {
        "x": 0,
        "y": 7
      },
      "reserved": false
    },
    {
      "id": "s08",
      "cords": {
        "x": 0,
        "y": 8
      },
      "reserved": true
    },
    {
      "id": "s09",
      "cords": {
        "x": 0,
        "y": 9
      },
      "reserved": true
    },
    {
      "id": "s011",
      "cords": {
        "x": 0,
        "y": 11
      },
      "reserved": false
    },
    {
      "id": "s012",
      "cords": {
        "x": 0,
        "y": 12
      },
      "reserved": false
    },
    {
      "id": "s013",
      "cords": {
        "x": 0,
        "y": 13
      },
      "reserved": false
    },
    {
      "id": "s014",
      "cords": {
        "x": 0,
        "y": 14
      },
      "reserved": false
    },
    {
      "id": "s12",
      "cords": {
        "x": 1,
        "y": 2
      },
      "reserved": false
    },
    {
      "id": "s13",
      "cords": {
        "x": 1,
        "y": 3
      },
      "reserved": false
    },
];