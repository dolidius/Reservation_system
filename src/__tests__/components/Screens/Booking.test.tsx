import Booking from '../../../components/Screens/Booking/Booking';
import { Provider } from "react-redux";

import * as renderer from "react-test-renderer";
import { createMemoryHistory } from 'history';

import { seats } from '../../../mock_data';

import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];

const mockStore = configureStore(middlewares);

const history = createMemoryHistory();

let store = mockStore({
    seats: {
        loading: false,
        seats,
        seatsToChoose: 0,
        nextToEachOther: false,
        error: false,
        errorMessage: "",
    },
});

describe("<Booking />", () => {

    it("should render correctly given seats data from store", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <Booking history={history} />
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

})