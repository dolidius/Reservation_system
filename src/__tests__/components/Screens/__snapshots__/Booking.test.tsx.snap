// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Booking /> should render correctly given seats data from store 1`] = `
<div
  className="sc-dIsUp dSxnkT"
>
  <div
    className="sc-bqGGPW kgDsjW"
  >
    <div>
      <div
        className="sc-eCApnc jYyZCg"
      >
        <div
          className="sc-bdnxRM sc-gtsrHT jsDwqN gtVyJb"
          onClick={[Function]}
        />
        <div
          className="sc-bdnxRM sc-gtsrHT ifDRTZ gtVyJb"
          onClick={[Function]}
        />
        <div
          className="sc-bdnxRM sc-gtsrHT dqnsBR gtVyJb"
          onClick={[Function]}
        />
        <div
          className="sc-bdnxRM sc-gtsrHT hrOdqx gtVyJb"
          onClick={[Function]}
        />
        <div
          className="sc-bdnxRM sc-gtsrHT dYaCah gtVyJb"
          onClick={[Function]}
        />
        <div
          className="sc-bdnxRM sc-hKFxyN cKTuvJ cIwBOA"
        />
        <div
          className="sc-bdnxRM sc-hKFxyN jiqDkV cIwBOA"
        />
        <div
          className="sc-bdnxRM sc-gtsrHT dOddcV gtVyJb"
          onClick={[Function]}
        />
        <div
          className="sc-bdnxRM sc-gtsrHT hhFeGh gtVyJb"
          onClick={[Function]}
        />
        <div
          className="sc-bdnxRM sc-gtsrHT koiqmx gtVyJb"
          onClick={[Function]}
        />
        <div
          className="sc-bdnxRM sc-gtsrHT egFetR gtVyJb"
          onClick={[Function]}
        />
        <div
          className="sc-bdnxRM sc-gtsrHT bVzAaN gtVyJb"
          onClick={[Function]}
        />
        <div
          className="sc-bdnxRM sc-gtsrHT gIDtnZ gtVyJb"
          onClick={[Function]}
        />
      </div>
      <div
        className="MuiBox-root MuiBox-root-1"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          Pozostałe miejsca do wyboru: 
          0
        </p>
      </div>
      <div
        className="sc-jSFjdj hVmDVn"
      >
        <div
          className="sc-fujyAs eWqvMj"
        >
          <div
            className="sc-bdnxRM sc-gtsrHT bXcdnt gtVyJb"
          />
          <span
            className="sc-pNWdM coaiQI"
          >
            Miejsca dostępne
          </span>
        </div>
        <div
          className="sc-fujyAs eWqvMj"
        >
          <div
            className="sc-bdnxRM sc-hKFxyN bXcdnt cIwBOA"
          />
          <span
            className="sc-pNWdM coaiQI"
          >
            Miejsca zarezerowane
          </span>
        </div>
        <div
          className="sc-fujyAs eWqvMj"
        >
          <div
            className="sc-bdnxRM sc-dlnjwi bXcdnt ilOmwR"
          />
          <span
            className="sc-pNWdM coaiQI"
          >
            Twój wybór
          </span>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKAaRy eALtjc MuiButton-containedPrimary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Rezerwuj
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
